dotnet/src/Connectors/Connectors.AI.OpenAI/TextToImage/AzureOpenAITextToImageService.cs

```diff
@@ -3,209 +3,173 @@
-    /// </summary>
-    /// <param name="operationId">The operationId that identifies the original text to image request.</param>
-    /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
-    /// <returns></returns>
-    private async Task<AzureOpenAITextToImageResponse> GetTextToImageResultAsync(string operationId, CancellationToken cancellationToken = default)
+    private static OpenAIClientOptions GetClientOptions(HttpClient? httpClient, int? maxRetryCount, string? apiVersion)
     {
-        var operationLocation = this.GetUri(GetImageOperation, operationId);
-
-        var retryCount = 0;
-
-        while (true)
+        OpenAIClientOptions.ServiceVersion version;
+        switch (apiVersion)
         {
-            if (this._maxRetryCount == retryCount)
-            {
-                throw new KernelException("Reached maximum retry attempts");
-            }
-
-            using var response = await this._core.ExecuteRequestAsync(operationLocation, HttpMethod.Get, null, cancellationToken).ConfigureAwait(false);
-            var responseJson = await response.Content.ReadAsStringWithExceptionMappingAsync().ConfigureAwait(false);
-            var result = OpenAITextToImageClientCore.JsonDeserialize<AzureOpenAITextToImageResponse>(responseJson);
-
-            if (result.Status.Equals(AzureOpenAIImageOperationStatus.Succeeded, StringComparison.OrdinalIgnoreCase))
-            {
-                return result;
-            }
-            else if (this.IsFailedOrCancelled(result.Status))
-            {
-                throw new KernelException($"Azure OpenAI text to image {result.Status}");
-            }
-
-            if (response.Headers.TryGetValues("retry-after", out var afterValues) && long.TryParse(afterValues.FirstOrDefault(), out var after))
-            {
-                await Task.Delay(TimeSpan.FromSeconds(after), cancellationToken).ConfigureAwait(false);
-            }
-
-            // increase retry count
-            retryCount++;
+            case "2022-12-01":
+                version = OpenAIClientOptions.ServiceVersion.V2022_12_01;
+                break;
+            case "2023-05-15":
+                version = OpenAIClientOptions.ServiceVersion.V2023_05_15;
+                break;
+            case "2023-06-01-preview":
+                version = OpenAIClientOptions.ServiceVersion.V2023_06_01_Preview;
+                break;
+            case "2023-07-01-preview":
+                version = OpenAIClientOptions.ServiceVersion.V2023_07_01_Preview;
+                break;
+            case "2023-08-01-preview":
+                version = OpenAIClientOptions.ServiceVersion.V2023_08_01_Preview;
+                break;
+            case "2023-09-01-preview":
+                version = OpenAIClientOptions.ServiceVersion.V2023_09_01_Preview;
+                break;
```

Comment:
VS highlights this case as not required, since it's the same as default option. Maybe we can remove it:
```suggestion
```

Reply:
personally I prefer being explicit, given that new versions will appear in time, but I don't care too much tbh, seems very subjective, unless there are build warnings...